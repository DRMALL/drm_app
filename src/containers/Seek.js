import React, { Component } from 'react'
import { View, Text, Image, ScrollView, TouchableOpacity } from 'react-native'
import TabBarItem from '../components/units/TabBarItem'
import SeekCategory from '../components/SeekCategory'
import SeekTab from '../components/units/SeekTab'
import { lightBlueColor, contentColor } from '../common/constants'
import { allParts, allTypes } from '../common/strings'
import { seek } from '../styles'
import { seekPartsData, seekTypesData } from '../utils/virtualData'

const seekIconSelected = require('../images/tabbar_icons/tabbar_search_selected_x.png')
    , seekIconNormal = require('../images/tabbar_icons/tabbar_search_normal.png')

export default class Seek extends Component {
  static navigationOptions = ({ navigation }) => ({
    tabBarIcon: ({ focused, tintColor }) => (
      <TabBarItem
        tintColor={tintColor}
        focused={focused}
        normalImage={seekIconNormal} 
        selectedImage={seekIconSelected} 
      />
    )
  });
  constructor(props) {
    super(props)
    this.state = (props => {
      let seekStateObj = {
        seekPartRow: false,
        seekTypeRow: false,
        selectedPart: allParts,
        selectedType: allTypes,
        topView: {position: 'relative', zIndex: 3},
        secondView: {position: 'absolute', zIndex: 2},
      }
      seekPartsData.map((partItem, indexp)=> {
        seekStateObj[`partColumn${indexp}`] = false
      })

      seekTypesData.map((typeItem, indext)=> {
        seekStateObj[`typeColumn${indext}`] = false
      })
      return seekStateObj
    })(props)
  }

  openModal(which) {
    this.setState({
      seekPartRow: which == 'seekPartRow' ? !this.state.seekPartRow : false,
      seekTypeRow: which == 'seekTypeRow' ? !this.state.seekTypeRow : false,
    })
  }

  pressPartColumn(p) {
    let partColumnOne = !this.state[`partColumn${p}`]
    seekPartsData.map((partItem, index)=> {
      if(p == index) {
        this.setState({
          selectedPart: partColumnOne ? partItem.parts : allParts,
          [`partColumn${p}`]: partColumnOne,
        })
      }
      else this.setState({[`partColumn${index}`]: false})
    })
  }

  pressTypeColumn(t) {
    let typeColumnOne = !this.state[`typeColumn${t}`]
    seekTypesData.map((typeItem, index)=> {
      if(t == index) {
        this.setState({
          selectedType: typeColumnOne ? typeItem.types : allTypes,
          [`typeColumn${t}`]: typeColumnOne,
        })
      }
      else this.setState({[`typeColumn${index}`]: false})
    })
  }

  render() {
    let selectPartRow = this.state.seekPartRow
      , selectTypeRow = this.state.seekTypeRow
      , topView = this.state.topView
      , secondView = this.state.secondView
    return(
      <View style={{paddingBottom: 80}}>
        <SeekTab partsData={seekPartsData} typesData={seekTypesData} state={this.state} openModal={this.openModal.bind(this)} />
        <View style={{height: '100%'}}>
          <View style={[!selectPartRow && !selectTypeRow ? topView : secondView, {height: '100%', paddingBottom: 50}]}>
            <SeekCategory {...this.props} />
          </View>
          <View style={[topView, {height: selectPartRow ? '100%' : 0}]}>
            <SeekPartsColumn partsData={seekPartsData} state={this.state} openModal={this.openModal.bind(this)} pressPartColumn={this.pressPartColumn.bind(this)}/>
          </View>
          <View style={[topView, {height: selectTypeRow ? '100%' : 0}]}>
            <SeekTypesColumn typesData={seekTypesData} state={this.state} openModal={this.openModal.bind(this)} pressTypeColumn={this.pressTypeColumn.bind(this)}/>
          </View>
        </View>
      </View>
    )
  }
}

const SeekPartsColumn = props => {
  let { partsData, state, openModal, pressPartColumn } = props
  return (
    <View style={{height: '100%'}}>
      <View style={seek.dataColumnView}>
        <ScrollView>
          {
            partsData.map((partItem, p)=> <TouchableOpacity key={p} 
              style={seek.itemTouch} 
              activeOpacity={0.8}
              onPress={()=> pressPartColumn(p)}
            >
              <Text style={[seek.itemText, state[`partColumn${p}`] ? {color: lightBlueColor} : {}]}>{partItem.parts}</Text>
            </TouchableOpacity>)
          }
        </ScrollView>
      </View>
      <TouchableOpacity activeOpacity={0.8} onPress={()=> openModal('seekPartRow')}>
        <View style={seek.halfOpacityView}/>
      </TouchableOpacity>
    </View>
  )
}

const SeekTypesColumn = props => {
  let { typesData, state, openModal, pressTypeColumn } = props
  return (
    <View style={{height: '100%'}}>
      <View style={seek.dataColumnView}>
        <ScrollView>
          {
            typesData.map((typeItem, t)=> <TouchableOpacity key={t} 
              style={seek.itemTouch} 
              activeOpacity={0.8}
              onPress={()=> pressTypeColumn(t)}
            >
              <Text style={[seek.itemText, state[`typeColumn${t}`] ? {color: lightBlueColor} : {}]}>{typeItem.types}</Text>
            </TouchableOpacity>)
          }
        </ScrollView>
      </View>
      <TouchableOpacity activeOpacity={0.8} onPress={()=> openModal('seekTypeRow')}>
        <View style={seek.halfOpacityView}/>
      </TouchableOpacity>
    </View>
  )
}
