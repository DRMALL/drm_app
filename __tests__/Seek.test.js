import 'react-native'
import React from 'react'
import Seek from '../src/containers/Seek'
import renderer from 'react-test-renderer';

test('Seek test', () => {
  const tree = renderer.create(
    <Seek />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})