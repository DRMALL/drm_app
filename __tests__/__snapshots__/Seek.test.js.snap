// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seek test 1`] = `
<View
  style={
    Object {
      "paddingBottom": 80,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "rgba(0,0,0,.06)",
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "rgba(0,0,0,.3)",
          "borderRightWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": "50%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(0,0,0,.6)",
              "fontSize": 14,
              "marginRight": 10,
            },
            Object {},
          ]
        }
      >
        所有配件
      </Text>
      <Image
        source={1}
      />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "width": "50%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(0,0,0,.6)",
              "fontSize": 14,
              "marginRight": 10,
            },
            Object {},
            undefined,
            Object {
              "opacity": 0.5,
            },
          ]
        }
      >
        所有类型
      </Text>
      <Image
        source={1}
        style={
          Object {
            "opacity": 0.5,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "position": "relative",
            "zIndex": 3,
          },
          Object {
            "height": "100%",
            "paddingBottom": 50,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F7F7F7",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0,0,0,.06)",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingVertical": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "rgba(0,0,0,.3)",
                  "borderRightWidth": 0.5,
                  "width": "28%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "rgba(0,0,0,.3)",
                    "fontSize": 14,
                  }
                }
              >
                物料长代码
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderColor": "rgba(0,0,0,.3)",
                  "borderRightWidth": 0.5,
                  "width": "26%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "rgba(0,0,0,.3)",
                    "fontSize": 14,
                  }
                }
              >
                物料名称
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderColor": "rgba(0,0,0,.3)",
                  "borderRightWidth": 0.5,
                  "width": "28%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "rgba(0,0,0,.3)",
                    "fontSize": 14,
                  }
                }
              >
                规格型号
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": "18%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "rgba(0,0,0,.3)",
                    "fontSize": 14,
                  }
                }
              >
                单位
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F7F7F7",
              "height": 500,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(0,0,0,.3)",
                "fontSize": 18,
              }
            }
          >
            暂无内容
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "position": "relative",
            "zIndex": 3,
          },
          Object {
            "height": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "opacity": 1,
            }
          }
        >
          <RCTScrollView>
            <View />
          </RCTScrollView>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "backgroundColor": "black",
                "height": "100%",
                "opacity": 0.5,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "position": "relative",
            "zIndex": 3,
          },
          Object {
            "height": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "opacity": 1,
            }
          }
        >
          <RCTScrollView>
            <View />
          </RCTScrollView>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "backgroundColor": "black",
                "height": "100%",
                "opacity": 0.5,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
